operators/audio/audio_decoder.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+
+#include <list>
+#define DR_FLAC_IMPLEMENTATION
+#include "dr_flac.h"
+#define DR_MP3_IMPLEMENTATION 1
+#define DR_MP3_FLOAT_OUTPUT 1
+#include "dr_mp3.h"
+#define DR_WAV_IMPLEMENTATION
+#include "dr_wav.h"
+
+#include <gsl/util>
+#include "string_utils.h"
+#include "string_tensor.h"
+
+struct KernelAudioDecoder : public BaseKernel {
+ public:
+  KernelAudioDecoder(const OrtApi& api, const OrtKernelInfo& info) : BaseKernel(api, info) {
+  }
+
+  enum {
+    DEFAULT_STREAM = 0,
+    WAV_STREAM,
+    MP3_STREAM,
+    FLAC_STREAM
+  };
+
+  int ReadStreamFormat(OrtKernelContext* context, const int64_t* p_data) {
+    int stream_format = DEFAULT_STREAM;
+    const OrtValue* ov_format = ort_.KernelContext_GetInput(context, 1);
+    if (ov_format != nullptr) {
+      std::vector<std::string> str_format;
+      GetTensorMutableDataString(api_, ort_, context, ov_format, str_format);
+      const char* formats[] = {"default",
+                               "wav",
+                               "mp3",
+                               "flac"};
+      for (auto fmt : formats) {
+        if (str_format[0] == fmt) {
+          break;
+        }
+        stream_format++;
+      }
+    }
+
+    if (stream_format == DEFAULT_STREAM){
+      auto p_stream = reinterpret_cast<char const*>(p_data);
+      std::string_view marker(p_stream, 4);
+      if (marker == "fLaC") {
+        stream_format = FLAC_STREAM;
+      } else if (marker == "RIFF") {
+        stream_format = WAV_STREAM;
+      } else if (marker[0] == char(0xFF) && (marker[1] | 0x1F) == char(0xFF)) {
+        // http://www.mp3-tech.org/programmer/frame_header.html
+        // only detect the 8 + 3 bits sync word
+        stream_format = MP3_STREAM;
+      } else {
+        ORTX_CXX_API_THROW("[AudioDecoder]: Unknown audio stream format", ORT_INVALID_ARGUMENT);
+      }
+    }
+    return stream_format;
+  }
+
+  template<typename TY_AUDIO, typename FX_DECODER>
+  static size_t DrReadFrames(std::list<std::vector<float>>& frames, FX_DECODER fx, TY_AUDIO& obj) {
+    const size_t default_chunk_size = 1024 * 256;
+    int64_t total_buf_size = 0;
+
+    for (;;) {
+      std::vector<float> buf;
+      buf.resize(default_chunk_size * obj.channels);
+      auto n_frames = fx(&obj, default_chunk_size, buf.data());
+      if (n_frames <= 0) {
+        break;
+      }
+      auto data_size = n_frames * obj.channels;
+      total_buf_size += data_size;
+      buf.resize(data_size);
+      frames.emplace_back(std::move(buf));
+    }
+
+    return total_buf_size;
+  }
+
+  void Compute(OrtKernelContext* context) {
+    const OrtValue* input = ort_.KernelContext_GetInput(context, 0);
+    const int64_t* p_data = ort_.GetTensorData<int64_t>(input);
```

Comment:
why `int64_t*` here? input 0 looks like it should be uint8 from CustomOpAudioDecoder::GetInputType()

Reply:
Good finding! Because all following usage are all type casted, the compiler didn't report issue.